The Identity System

JJRcop opened the pull request:
:cl:
add: Adds the identity system, which changes the way you see names.
add: Instead of instantly knowing everyone's name, you have to learn their names. This can be done by examining them, or manually setting their name with the new identity manager. <b><a href='https://tgstation13.org/wiki/Identity'>More info here.</a></b>
/:cl:

You no longer see everyone's names instantly when they speak, what you see is dependent of how you remember their name.

Important bits: 
* Department peers [know among themselves](https://github.com/JJRcop/tgstation/blob/1ac24e4f6d87f5be58b2c370010604631f8b5ed8/code/modules/jobs/jobs.dm#L56-L136) at roundstart. As do initial team antagonists.
* There is a configurable (default 60 second) "retcon limit" where latejoins that are in before the limit will have their identities backfilled in as if they readied up roundstart.
* It's a little bit difficult to run around and know everyone by examining everyone you see, as you will just know their faceprint unless they spoke in front of you within 50 seconds before. And if you're still persistent, this mimicks being hyper paranoid in real life.
* Silicons get their identity info directly from crew records.

I recommend you read the [ugly wiki article](https://tgstation13.org/wiki/User:JJRcop/Identity) I wrote as that is more comprehensive (Hopefully it will be rewritten by someone who knows what they are doing :))

### DO NOT MERGE
**until the code for the system itself is approved and I retrofit all output that deals with mobs so that it works with the identity system.
And especially logging.**

I may need to get a temporary freeze worked out a'la #20793 during the retrofitting process

- [X] Complete code for identity system (subject to change from feedback)
- [ ] Code approved
  - [ ] examine needs to be better
  - [x] click on names in chat to rename them, don't use the identity manager for this, but keep it around still
  - [x] make the names in chat not blue all the time so that accents are visible
  - [x] list lengths should be defines
  - [x] move global regex vars to be static vars in the procs
  - [x] raz doesn't like var/_these_vars
  - [ ] limited range for faceprint gathering?
  - [x] config file for pre-knowns
  - [x] add accents
  - [ ] some form of help
  - [ ] potential garbage issues for identity manager
- [ ] Retrofitting for logging
- [ ] Retrofitting for visible_message, show_message
- [ ] Retrofitting for <<

An additional refactor: All mob/livings can now whisper, and the human whisper proc has been removed. If this PR gets denied in the end I will extract that refactor and PR it on its own, please do not consider this refactor when making the decision on whether to merge this, as it was something that I had to do in order for this to work, but can easily work on its own. I do not want to separate them at this time, though.

Commit:
Adds base identity system code
TODO: Retrofit all output that deals with mobs

Kromgar commented:
Oh my god this is incredible.

KingPhilipIII commented:
Disregarding the fact that as Employees of Nanotrasen we should all know each others names.

And we wear FUCKING IDS.

SweaterKittens commented:
What about other players that you will reasonably know? This will make it nigh-impossible to forge meaningful connections with other characters in-game, since everything resets.

Kromgar commented:
@SweaterKittens You just need to meet them and examine pda them or something

JJRcop commented:
Shush I wanted to PR this ASAP I'm editing the PR body with more info right now

Haevacht commented:
You'll recognise your metabuddies's outfits and hair.

SweaterKittens commented:
@Kromgar So the person who I roleplayed with as my character's wife over the course of a year - I wouldn't know what their voice sounds like? What about people who roleplay relatives? Or coworkers who have worked together time after time?

XDTM commented:
Does anyone actually do that? And isn't it an issue with open metafriending?

SweaterKittens commented:
People do, yes. Typically the latter: coworkers who know each other because they work in the same department. Metafriending only becomes a problem when it gets wildly out of hand - e.g. people giving each other all-access every round, people self-antagging to fight for their metafriend, etc etc. Encouraging roleplay and the unique interactions that stem from fostering relationships with other crewmembers is never a bad thing in moderation. But as I said, this will completely kill what little roleplay we have left and force silent drive-by-examines and 0 interaction.

XDTM commented:
Should be acceptable if you start with your department and Heads of Staff as known people roundstart.

TehZombehz commented:
RIP Roleplay.

JJRcop commented:
I could always figure out a system that allows persistent relationships.

SweaterKittens commented:
@JJRcop If you did that, and perhaps made it so you can't drive-by everyone on the station at roundstart, I would be 100% for this. I definitely don't mean it as an offense to this idea or you personally when I take issue with this - we just have very little roleplay left and I would hate to see it be killed entirely.

anconfuzedrock commented:
its already dead if the only roleplay we have is a few snowflakes stalking each other

JJRcop commented:
The main intent behind this feature was to help create situations where roleplay could much more easily come about, and to also disrupt the gameyness of /tg/station. Actually mostly disrupt the gameyness part. So if this thing somehow kills roleplay then it will be doing the exact opposite of what my intentions were with it.

SweaterKittens commented:
@JJRcop I don't want to totally derail your PR, so perhaps this is a conversation best had over IRC - but the jist of it is that one of the main flaws with the game that Kor pointed out is that there's too much information available to the player. While this attempts to stifle that to a degree, there's still the fact that you can shift-click on someone all the way across the screen while you're running and get a detailed rundown about who they are and everything they have on them.

KorPhaeron commented:
This sounds like it could be interesting in a smaller population but I'd imagine it scales terribly to 60-80 people screaming on the radio and you have no idea who any of them are

Steelpoint commented:
This is a very .... interesting change.

It has the capacity to have a major impact on the way rounds progress, at least early on.

However all I can initially see this system doing is having every player spend the first ten minutes shift+clicking everyone they see.

It has the capacity to be tedious busy work.

RemieRichards commented:
I've said this to you before JJ, the idea is just flawed in concept, it's not going to work.

francinum commented:
This is a terrible idea, and I believe you've been told this about 20 times since you had this idea.

francinum commented:
Also, what of the AI, who can't examine people.

JJRcop commented:
Silicons have their identity information taken from crew records

Steelpoint commented:
I see what you're trying to accomplish and its a commendable effort.

At best however, I think this system may be better served to only apply to certain off station antagonists. It would make sense for Nuclear Operatives, Emergency Response Teams or for Wizards to not know the names of most of the members of the crew. 

Perhaps Changelings would also not initially know the names of anyone on the station,

But that's at best. 

While I do in theory support this idea, but I can't support it in practice as it stands.

RussellTalbott commented:
## Captain Announces

Everyone to the bar for the daily meet and greet.

And correct me if I'm wrong but the shift doesn't start with 60 strangers teleported onto the station at once it's just this specific shift that goes to shit or maybe I'm wrong

dannno commented:
I'm simply left wondering why.
What is the value that makes this worth adding?

and
>tgui

Supermichael777 commented:
is it based off id name when first met?

FantasticFwoosh commented:
Lizard names should be unintelligible to human crew. And a ability to personally nickname a future voice you can hear (ei - "_shitty lizard captain_" says: sstop breaking into my office!!) would also help clear up confusion.

If we're going to go down the metafriend route, why not just have OOC in-client friend/buddy requests for station roles know each other as a prequisite? Then reasonable meta-cliques of non-antag players (with rules for traitors/lings as needed) can be monitored.

Injazz commented:
Well now we now this RolePlay thing was a mistake and i happy we about to come over it.

Armhulen commented:
Make heads know all the people on the ship. or at LEAST all in their department. greyshits should be unknown and all lizards should be too (unless security)

chap is known for preaching, everyone knows him and he knows everyone.

Iamgoofball commented:
if anything this encourages roleplay because you gotta talk to fuckers to find out who they are

can we get a testmerge done?

Thunder12345 commented:
10 minutes of busywork every round start as you try to work out who everyone is. :-1: x 💯

dannno commented:
this doesn't encourage roleplay it just adds tedium

Iamgoofball commented:
all I hear is babies whining about having to talk to people

francinum commented:
>this doesn't encourage roleplay it just adds tedium

Thunder12345 commented:
Goof do you want to have to learn who everyone is every single round?

Iamgoofball commented:
yes

Thunder12345 commented:
Even the people who intentionally never speak or wear their ID so you can't identify them when you see them murderboning?

Iamgoofball commented:
Yup

Thunder12345 commented:
Sounds like a good reason never to speak, to maximise powergame by being unidentifiable

Iamgoofball commented:
And then you won't ever be saved when you call for help because no one knows who you are or what you look like.

Thunder12345 commented:
As long as you give out your location you've got as much chance of getting help as ever

Iamgoofball commented:
Or you give out your location after making friends with a few crewmembers and they come help you.

Thunder12345 commented:
And why is not knowing anybody going to make people any more likely to make friends than before?

Iamgoofball commented:
Because your chances of survival drop insanely.

I don't know what kind of social introvert you have to be to not see the benefit in /talking to people/.

Thunder12345 commented:
An admin who understands how this server's playerbase operates

PKPenguin321 commented:
This is stupid and aside from the points that were already made this will make admin investigations an absolute bitch when nobody can name anybody else over ahelps

Heavy :-1:

dannno commented:
Oh my god this is so much worse in practice than I ever could have imagined. What the fuck were you thinking?

francinum commented:
I should have brought that up awhile ago, yes, this would make admin investigation bleeding hellfire because you'd have to ask around to figure out who 'Person in the CE suit' is.

dannno commented:
![](http://i.imgur.com/yXsS62I.png)

Armhulen commented:
okay that's pretty shit, :-1:

francinum commented:
WE TOLD YOU THIS WOULD HAPPEN.
### **_YOU DIDN'T LISTEN_**

PKPenguin321 commented:
CLOSE this PR

lzimann commented:
Ghosts can't see the name of people(even if they examine the person).

Iamgoofball commented:
S
A
L
T
ghost is valid thouhg

dannno commented:
There is a reason even HIGH-RP servers don't have things like this. You should have thought about that before putting effort into this.

YourNeighborsObnoxiousCat commented:
10/10 feature.
![badpull](https://cloud.githubusercontent.com/assets/25470377/22621814/d6a9c298-eaf1-11e6-8bfc-44f22298f889.png)

Thunder12345 commented:
Just one round was enough to prove that this was a horrible idea

KorPhaeron commented:
>people acting retarded on purpose 
>this proves something is bad

I don't think this is a good idea but it still irritates me that people ruin the round on purpose because they're angry at new ideas

lzimann commented:
Another big issue: Admins will see attack logs as "Person(ckey)", can be kinda annoying to deal with.

francinum commented:
It's an example of what can happen.

RussellTalbott commented:
Having to fucking manually attach names to faces like what the fuck is this let me just got to the bar and get a GOOD LOOK at all these retards HEAR THEIR VOICES and then SIT MYSELF DOWN AND FUCKING ATTACH VOICES TO FACES MMMM YUMMY

KorPhaeron commented:
@francinum 

Well if everyone started spamming or randomly attacking each other the game would break down under our current system as well

francinum commented:
I was under the impression they did that anyway.

KorPhaeron commented:
@lzimann 

LeoZ: is is good to testmerge

JJRcop: um, probably not
JJRcop: since all actions that aren't speech will show as Person attacks Person

He told you not to testmerge it you already knew about this

dannno commented:
Idea sounds good on paper. But this is simply unreasonable, even without the bugs.

AnonymousNow commented:
This is a bad idea. I want more RP back on the servers, but this is not the way to go about it.

Iamgoofball commented:
@JJRcop 
Make it so if I say my name or part of my name(minimum 3-5 words), it automatically links anyone near me with that name.

scoopscoop commented:
Game is broke pls unmerge

Thunder12345 commented:
@Iamgoofball have you seen anybody RP because of this system yet?

dannno commented:
Don't bother, goofball is in legitimate denial and trapped in a delusion

factoryman942 commented:
I agree with making this apply to off-station originating antags, but even then you'd have to have some way to recognise different people speaking - so Unknown A, B, C, etc

francinum commented:
Where the hell even IS the management button.

Jalleo commented:
People. You seriously shouldnt test merge until @jjrcop thinks its ready to try. And secondly:

This is actually going to be a good idea LONG TERM short term we find all of those that are rubbish and clean them out. Seriously it forces RP and once this is truly in this will begin a move to making RP happen. The other thing it will require is a good (semi) randomised character creation to lower meta chance while not having fucking awful names and looks. Once that is all on we can truly begin to start making more mechanics and design to make suitable RP as attitudes change.

This is a change like goof medicine. People dealt with it in the end just like the change to engineers spawning in with insulated gloves getting removed.

JJRcop commented:
I mean the idea was you guys would check on the code first BEFORE I touch nearly every file in the game to get rid of the Person gives Person a swirly. But thanks for the notice about ghosts, forgot about that.

dannno commented:
I have no idea where this "It forces RP" shit is coming from, running around spamming shift click isn't RP

francinum commented:
This, as it stands, is annoying tedium and does nothing to """Force RP""" besides piss people off.

FantasticFwoosh commented:
Plasma men shouldn't be able to be identified at all within their space suits, especially if you made the face mask more opaque. To counteract this you could probably have the plasma-man's name stencilled onto the owner of the suit, but without *knowing* them you wont associate a voice to the suit.

Dorsisdwarf commented:
I like this, but if you dont know the heads and your own department it would suck dick.

BeeSting12 commented:
As other people have said, forced tedium is not RP.

RemieRichards commented:
I think you've amazingly miscalculated how well this would go with the fact that you've already written and published a wiki page.

factoryman942 commented:
it's more drafting a wiki page than writing and publishing it, really

KorPhaeron commented:
@tgstation/commit-access 

goof asked people from goon and hippie to come upvote this PR so remember that the tally up there is more or less worthless for gauging reaction now

PKPenguin321 commented:
i asked people on sybil to come downvote the PR hohoho

RemieRichards commented:
@factoryman942 not when you can just go to the wiki, type in "Identity" and actually obtain this page.

optimumtact commented:
If anything we should encourage more developers to write wiki pages (in their own namespace of course) for their pr'ed features before they are even pr'ed.

RemieRichards commented:
@optimumtact it lends unwarranted legitimacy and authenticity to content that isn't in the game (any may never BE in the game)

optimumtact commented:
I don't see how that would affect if the pr got merged or not, other than making me look more favourably upon the developer for bothering to document their feature ahead of time and showing they have put a touch more effort and thought into things.

anyway it's aside to the pr  and is better discussed in the coding forum, I just thought I'd mention it in passing because I dont' want to discourage people from drafting wiki pages for their features under their user namespace.

RemieRichards commented:
@optimumtact I never said it would affect it's merger, I simply said JJ was quite wrong to assume his feature would get in, which bothering to produce a wiki page suggests he intends that page to be used, and thus that the feature will be added, as in it's factually true or probably likely it will be, which is entirely wrong of him to assume.

Steelpoint commented:
It is no effort to remove a wiki page, and I would rather encourage than discourage coders to write wiki content on their creations.

Worse case you can ask them to write content on their user profile page so no one accidently thinks they are writing a page for content already in the game.

Iamgoofball commented:
hey at least he wrote a guide on how to use it

Grazyn commented:
You can't read paper unless you're one tile from it, for consistency you shouldn't be able to read the name on the ID unless you're standing next to them.

JJRcop commented:
I did not assume that this would be put in, if this gets denied I'll remove the wiki page, and I'll actually put it in my userspace for now.

Cheridan commented:
>Again, even the most autistic RP servers don't have anything like this. That MEANS something.

Uh, all that it means is that they can't coded it. You could make that statement about literally every feature that doesn't exist yet.

JJRcop commented:
Oh um, please review this.

francinum commented:
Public opinion should be telling enough.

Iamgoofball commented:
why was my post deleted here

Commit:
Moves some procs into identity.dm for now
Fixes getpois to use real_name
Moves human examine identity code into get_interact_name proc
Overwrites this for silicons to use crew records and ghosts to always see real_name
Overwrites most identity procs for ghosts to just shortcut to real_name
Fixes parse_identity_subjects so it doesn't require every subject to be in the string
Makes hugging and picking someone up use the identity system as an easy way to test it

JJRcop commented:
So I fixed the ghost thing, good catch, added the identity system for dialog to hugging for demonstration purposes, and patched up the rest of the silicon crew records thing so it works for all the identity stuff, and some other little things, on top of rebasing again so that while I'm testing this locally I have the new SDQL improvements and runtime bugfixes :D

Commit:
Fixes silicon pictures not transferring mobinfo correctly

Dorsisdwarf commented:
I'l be honest.

Since when has coderbus given the slightest of fucks about overarching deeply-unpopular changes to the core gameplay of ss13? 'The players like it/dont like it' has universally only ever been taken into account when it supports the changes.

So why does everyone here suddenly pretend to care about that sort of thing, when past experiences have shown they really dont?

francinum commented:
Because this takes overarching change to a massive extreme, by changing the most fundamental thing: knowing who people are.

Dorsisdwarf commented:
You mean like when Paprika halved every single stun in the game without consulting anyone?
Combat is just as fundamental.

How about runspeed, the least popular change of all time. Freedom of movement is the absolute core of SS13's combat system, but that got forced through anyway.

Why do people pretend to care now?

optimumtact commented:
all of those changes were consulted with teh only person who matters, me

JJRcop commented:
Hey could I get the WIP tag removed? This is done and I am ready for review by a maintainer. I don't want to go through the retrofitting process and constantly deal with merge conflicts until this is approved. If you need more time to discuss the concept itself aside from the code let me know.

JJRcop commented:
Hey, can I get the WIP tag removed?

PKPenguin321 commented:
I really hope this does not get merged.

Steelpoint commented:
Dorsisdwarf is right in that in the past coderbus has never really given a fuck about what the players want, the only concerns in merging PRs, at least when I was last active, was appeasing the head coders, meaning Cheridan and whomever else was on top that day.

Its how people like Paprika were enabled to do whatever they want.

I don't know about now days. It seems like oranges is in charge, or at least is massively involved in the merging of content PRs.

KorPhaeron commented:
I feel like this is the kind of thing that would actually make people quit if merged

JJRcop commented:
Funny thing I ask for the WIP tag to be removed and here I am fixing a few issues I hadn't noticed before. I'll just be a second.

Commit:
Fixes some AI tracking issues

JJRcop commented:
Okay there, seriously ready now

Iamgoofball commented:
@KorPhaeron 
People quit over goofchem.
People quit over stun nerf.
People quit over movement speed slowdown.

We still have a server.

RemieRichards commented:
@Iamgoofball Doesn't mean we should have this moronic idea.
@Dorsisdwarf Doesn't mean we can't start caring.

I stand by this being nuked from orbit.

Steelpoint commented:
At least goofchem, the stun nerf and the movement speed changes did not change the fundamental core basics of the game, being able to interact with other people via communication.

This is a massive change to something that EVERYONE will be affected by, and all I see this system doing is adding a useless check list of shift+clicking everyone you see.

Security is going to be further screwed over by this, how do you arrest someone you can't even name? What if they are not wearing a ID badge so good luck. Better strip search literally everyone you see to make certain they are not someone in disguise.

AnonymousNow commented:
I personally believe that this is a very bad idea that would serve only to cause confusion and not aid roleplay.

We should make a poll.

Iamgoofball commented:
Remie don't be a dick

Grazyn commented:
>Security is going to be further screwed over by this, how do you arrest someone you can't even name?

Officers have security records, with names and photo IDs. Setting someone to arrest will still show you the wanted icon on the sechuds when you see them, as long as they're wearing the ID. Yes, they can change their job uniform, cut their hair, hide the ID and their face, but everyone can already do this. How's that different from, let's say, a chemist killing someone, then wearing a greyshirt and box helmet without the ID?

francinum commented:
"Someone beat my head in with a wrench."
"Who was it?"
"How the fuck am I supposed to know that."

PKPenguin321 commented:
Please help. These randoms keep trying to PR awful gamewrecking things to my favorite game and maintainers wont just close it. What do I do?

Grazyn commented:
>"Someone beat my head in with a wrench."
"Who was it?"
"How the fuck am I supposed to know that."

But this happens already. Traitor jumps you with no ID, face covered, and random suit he will change soon after the crime. "Who was it?" "How the fuck am I supposed to know that."
Or, he doesn't bother hiding his identity. With the current system: "Who was it?" *looks at the attack logs* "Griffy McDoctor."
With the new system "Who was it?" "A doctor, with blonde hair. He came through that door." "Hm let's check the records. We have 3 doctors, was it him? Look at the photo." "Yes it was him!" ":h Set McDoctor to arrest. And take the prints from the door, just in case we need to double check the ID." 

B-but this will totally break RP!

KorPhaeron commented:
We should make attack logs vanish as they scroll up debate me

RemieRichards commented:
This has until I get home (9 ish hours from now) to get a full code review.

If it's reviewed, and it's ok, then we can discuss further whether it should be added.
If it's reviewed and it's shitcode, you can have more time, that's fine.
if it's not reviewed, nobody cares enough and I'll close it.

The fact of the matter is, in 5 days, nobody has reviewed this, the idea just doesn't mesh with any of the people who actually review and make the decisions to merge or not (either that or they're being shy), add onto that a huge backlash from people here (vocal minority perhaps, but the otherside have had time to speak aswell) and it just doesn't seem like it fits.

My opinion remains the biggest "no" I could ever give.

Thunder12345 commented:
@Iamgoofball That's because the space left by the older players was filled with hubbies

Grazyn commented:
@RemieRichards Looking at the number of upthumbs/downthumbs, it doesn't seem like this is so universally hated like you seem to imply. I've read the "backlash" from the "vocal minority" and most of the critics boil down to:

>It will kill metrafriendships

Even assuming this is bad, how would it kill them exactly? If your friends uses static character appearance, you still know how they look like. You can still send them a pda message. The only difference is that you have to examine them, or talk to each other to link the name. Something that you would already be doing since you're metafriends. If your only interaction with your metabuddy during a round is to silently rush to aid them whenever they call for help on the radio and otherwise ignore them, well yes, I can see how this would slightly incovenience you.

>It will kill roleplay

Giving people more reason to talk to each other definitely doesn't kill roleplay.

>Everyone running around at roundstart drive-by examining people isn't roleplay

That's why it's gonna be tweaked to work only if you're 2 tiles from them. After a while you realize you don't actually need to know the names of ~80% of the crew you will never interact with anyway during a regular round.

>You won't be able to tell who's attacking you

You already can't if they bother to hide their identity. And if they don't, all it takes is 0,5 seconds to shift-click them. Still better than having to manually copypaste their name from attack logs to IC notes because Kor actually managed to make attack logs vanish after some time. And gives more depth to police work. Makes people actually pay attention to what happens to them.

>The coder wrote a wiki page detailing this new game mechanic. The arrogance, the gall!!

I don't think I need to comment this

RemieRichards commented:
@Grazyn if you'll read above, your "ally" goofball completely invalidated the thumb count by asking goons (who's opinion is not important for a /tg/ change) to spam thumbs, my talk of hatred and backlash came from the comments which are either "Oh god this is shit" or "I like the idea but I don't think it'll work as is"

AnturK commented:
I'm not really sold on the idea, if a feature should introduces tedium it's either for balance reasons or in exchange for new ways of interaction, this doesn't really produce any new situations, masked person will still not say anything because they have no way of knowing if someone recognizes their voice, and for other interactions it's mostly irrelevant ? I might be missing something obvious here.

PKPenguin321 commented:
How the fuck will the radio work with this? I feel like it will become unusable

Iamgoofball commented:
Did JJR piss in your cereal or something? I'll reopen this if it gets closed for stupid reasons.

PKPenguin321 commented:
No but he's currently trying to take a shit on my favorite game

RemieRichards commented:
@Iamgoofball No, my working relationship with JJ is fine, this is just a really dumb idea.

JJRcop commented:
I am not going to retrofit the entire codebase until the base system itself is approved and we decide to either testmerge it or completely merge it. I already retrofitted a few spots already as an example of how it will go, for the review process, if I retrofitted the entire codebase in this PR right now I'd have to deal with a lot of conflicts, and it would have a whole lot more files changed, which wouldn't be good for anyone.

@RemieRichards If you wouldn't mind, I ask to give it a little while longer.

JJRcop commented:
Also maintainers, can you at least review the code? I know we're still discussing the concept itself, that's fine, I knew this would take a while, and I'm prepared for that, but it would help so much if the code was verified as good at least so we can discuss changes I can make to this to make it more acceptable.

Grazyn commented:
> Writing the wiki page isn't arrogance, releasing it publicly on the actual wiki is, the wiki only holds content that IS or WAS part of the game, placing it there lends it unwarranted legitimacy.

He moved it to a user page as soon as you pointed out the issue, this doesn't look like it was done with malice to me, unless you have a past history with him that would let you think otherwise. We had blatant grudgecode and salty PRs merged or at least seriously taken into consideration in the past, using this wiki page thing as a reason to close this PR is simply ridiculous.

JJRcop commented:
@Grazyn It's fine, really. No need to argue with Remie. She isn't closing it due to the wiki page, she's closing it because nobody wants to review this. I would prefer to have it open a little bit longer, though.

Iamgoofball commented:
I'll review it if I have to

kevinz000 commented:
Pls stop.

Cyberboss commented on code/modules/mob/identity.dm:
```diff
@@ -0,0 +1,578 @@
+
+var/global/list/used_voiceprints = list()
+
+/proc/generate_voiceprint()
```
Make it static if it's used nowhere else

Cyberboss commented on code/modules/mob/identity.dm:
```diff
@@ -0,0 +1,578 @@
+00 [3] Name associated with voiceprint. IDENTITY_INTERACT is semi-permanent and IDENTITY_MANUAL is permanent, wereas
+00    -- IDENTITY_HEARD and IDENTITY_SEEN are not.
+00 [4] Associated faceprint if available.
+00 [5] Last message heard.
```
No. No. No. No. NO! These ALL NEED DEFINES!

Cyberboss commented on code/modules/mob/identity.dm:
```diff
@@ -0,0 +1,578 @@
+		AM_temp = AM_identity[5]
+		AM_temp_time = AM_identity[6]
+	else
+		AM_identity = new(6)
```
Even this 6 needs to be a define

Cyberboss commented on code/modules/mob/identity.dm:
```diff
@@ -0,0 +1,578 @@
+	var/cat = CATEGORY_VOICEPRINTS
+	var/datum/mind/mind
+	var/list/print_refs = list()
+	var/list/refs_lookup = list()
```
Init these in New or preferably lazy init them

Cyberboss commented on code/modules/mob/identity.dm:
```diff
@@ -0,0 +1,578 @@
+	if(the_mind && istype(the_mind))
+		mind = the_mind
+	else
+		qdel(src)
```
I don't this is GC safe. Regardless of what `Destroy` does, the qdeleted datum will be assigned somewhere. Unless you are just creating it with `;new /datum/identity_manager(x);`. Anything that creates one of these will have to check `if(QDELETED(x))`. Instead, you should have New CRASH if there is no mind datum and have callers assert that for you

RemieRichards commented:
@Iamgoofball as someone who tends to produce code that fails to meet standard fairly often, no, you won't (as in you're welcome to waste your time but your review will be discarded)

@JJRcop you can have another week, that's fine.

@Grazyn I'm criticising it primarily as it's never (to my knowledge) occured before, it's either:
* No wiki page
or
* A wiki page after the feature is added
If it was purely a mistake (which as you say he's corrected) that's fine, notice how I haven't brought it up unwarranted besides my initial comment?

Cyberboss commented on code/modules/mob/identity.dm:
```diff
@@ -0,0 +1,578 @@
+/datum/identity_manager/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = 0, datum/tgui/master_ui = null, datum/ui_state/state = alive_state)
+	if(!mind)
+		qdel(src)
+		return
```
Again, unless you can guarantee Destroy will clean up references, this is probably not safe

Cyberboss commented on code/modules/mob/identity.dm:
```diff
@@ -0,0 +1,578 @@
+		return
+	if(!mind)
+		qdel(src)
+		return
```
Same thing

Cyberboss commented on code/modules/mob/living/carbon/human/say.dm:
```diff
@@ -1,12 +1,10 @@
-/mob/living/carbon/human/say_quote(input, spans)
-	if(!input)
-		return "says, \"...\""	//not the best solution, but it will stop a large number of runtimes. The cause is somewhere in the Tcomms code
```
Are you sure it's safe to get rid of this?

JJRcop replied:
Yes, it's safe. I tested it

JJRcop commented:
@Cyberboss said in IRC that at the very least this could be implemented in a "disabled state", which would mean I just make everyone know everyone in the system. I'm fine with that if that's necessary. I'm also OK with the "this is probably good if it's only external antags that don't know the crew". Those paths are both fine with me, and it's extremely easy to make it that way.

Right now, I need a maintainer to say that the code is OK, or point out things I should change code-wise, then I will take the feedback and make changes.

Cyberboss commented on code/modules/mob/living/say.dm:
```diff
@@ -1,64 +1,64 @@
 
 	  //kinda localization -- rastaf0
 	  //same keys as above, but on russian keyboard layout. This file uses cp1251 as encoding.
-	  ":ê" = "right hand",	"#ê" = "right hand",	".ê" = "right hand",
```
Why were these changed?

JJRcop replied:
# is used for whispering now so I removed the #radio codes

Cyberboss commented:
That also allows downstreams to fuck with it if they want

RemieRichards commented:
I don't agree with the code going in even in a disabled state, it'll simply make it harder to remove as time goes on, a reminder that Ninja only got it's filthy tendrils removed from 80% of the codebase because I got home from college and had an IRL version of a dorf's fey mood.

Any parts of the system that may actually be of future use (for example any parts of the code made cleaner by this (say code voiceprints?)) I'm happy to allow in if they're up to standard.

Cyberboss commented on code/modules/mob/living/say.dm:
```diff
@@ -96,18 +98,33 @@ var/list/crit_allowed_modes = list(MODE_WHISPER,MODE_CHANGELING,MODE_ALIEN)
 
-	if(message_mode != MODE_WHISPER) //whisper() calls treat_message(); double process results in "hisspering"
-		message = treat_message(message)
+	var/message_range = 6
```
This was originally 7 wasn't it?

JJRcop replied:
Yes, but since you can eavesdrop on everything 1 tile above the specified range, so I made it 6 so that when you "eavesdrop" you're still on screen.

Cyberboss commented on code/modules/mob/living/silicon/ai/say.dm:
```diff
@@ -49,7 +56,7 @@
 	var/obj/machinery/holopad/T = current
 	if(istype(T) && T.masters[src])//If there is a hologram and its master is the user.
-		send_speech(message, 7, T, "robot", get_spans())
+		send_speech(message, 6, T, "robot", get_spans(), get_voiceprint())
```
Again

Cyberboss commented on code/modules/ninja/suit/suit.dm:
```diff
@@ -21,6 +21,7 @@ Contents:
 	resistance_flags = LAVA_PROOF | ACID_PROOF
 	armor = list(melee = 60, bullet = 50, laser = 30,energy = 15, bomb = 30, bio = 30, rad = 30, fire = 100, acid = 100)
 	strip_delay = 12
+	identity_name = "Space Ninja"
```
Probably should be just ninja, no?

JJRcop replied:
SPAAAAAAAAAAAAAAAAACE

Cyberboss commented on code/modules/paperwork/photography.dm:
```diff
@@ -234,43 +258,44 @@
-	var/mob_detail
+	var/list/mob_detail = list()
 	for(var/mob/M in the_turf)
+		var/list/M_info = new(3) //[1] = if it's a faceprint or just a direct name, [2] = for the faceprint or direct name, [3] = looks hurt or not, [thereafter] = stuff they are holding
```
D E F I N E S

Cyberboss commented on tgui/src/components/display.ract:
```diff
@@ -1,7 +1,7 @@
   {{#if title}}
     <header>
-      <h3>{{title}}</h3>
+      <h3>{{{title}}}</h3>
```
Necessary why?

JJRcop replied:
Otherwise it won't parse the HTML tags in title, which I use in the identity manager

RemieRichards replied:
```{{}}``` to import text
```{{{}}}``` to import HTML/JS
If you're going to comment on moustache templates @Cyberboss, atleast understand the syntax.

Iamgoofball commented:
Stop being a dick, Remie.

Cyberboss commented on tgui/src/interfaces/radio.ract:
```diff
@@ -15,6 +17,11 @@
         action='listen'>
         {{data.listening ? "On": "Off"}}</ui-button>
     </ui-section>
+	{{#if data.headset == 2}}
```
When you make this a define, put a comment here

Cyberboss reviewed (changes requested):
I'm not letting this get closed because it didn't get reviewed. I want this merged but, I'm not easy either.

This DESPERATELY needs defines

Thunder12345 commented:
Because obviously nobody could ever disagree with a feature you support, Goof?

Iamgoofball commented:
Seriously thinking this needs headcoder protection

RemieRichards commented:
You are an absolute moron goof.

francinum commented:
What does this add to the game besides annoyance and something to later
dike out because it was hated.
On Thu, Feb 9, 2017 at 13:39 Iamgoofball <notifications@github.com> wrote:

> Stop being a dick, Remie.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/tgstation/tgstation/pull/23693#issuecomment-278732969>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFUGuMbg_74FysG1EnKuigda04QQMe6-ks5ra110gaJpZM4L3KrT>
> .
>

Thunder12345 commented:
If nobody can make negative comments, nobody dislikes the PR!

dannno commented:
Christ sake goofball turn your brain on

JJRcop commented on code/__HELPERS/type2type.dm:
```diff
@@ -437,49 +437,42 @@ for(var/t in test_times)
+var/global/list/colors_list = list("white" = "#FFFFFF", "black" = "#000000", "gray" = "#808080", "brown" = "#A52A2A", \
+									"red" = "#FF0000", "darkred" = "#8B0000", "crimson" = "#DC143C", "orange" = "#DC143C", \
+									"yellow" = "#FFFF00", "green" = "#008000", "lime" = "#00FF00", "darkgreen" = "#006400", \
+									"cyan" = "#00FFFF", "blue" = "#0000FF", "navy" = "#000080", "teal" = "#008080", \
```
they don't, easier to read though because otherwise the line extends into outer space

Iamgoofball commented:
Have you guys considered trying the finished version before making snap decisions

RemieRichards commented:
>the finished version of a flawed concept is somehow not flawed

Umm, No?

Iamgoofball commented:
Do I have to PR stun gloves to prove a point that players don't have the best opinions

JJRcop commented on code/__HELPERS/unsorted.dm:
```diff
@@ -291,10 +291,7 @@ Turf and target are seperate in case you want to teleport some distance from a t
-		var/name = avoid_assoc_duplicate_keys(M.name, namecounts)
-
-		if(M.real_name && M.real_name != M.name)
-			name += " \[[M.real_name]\]"
```
Person [Gregory Stratton]
Person [Alfonzo Herman]
Person [Mary Smith]

RemieRichards commented:
Your PR will be immediately closed because you're doing it out of spite. so go on, make all of our days.

JJRcop commented on code/_onclick/hud/screen_objects.dm:
```diff
@@ -83,6 +83,22 @@
+/obj/screen/identity_manager/Click()
+	var/mob/M = usr
+	if(isobserver(usr))
+		return
```
The observe verb shows you the subject's UI

Steelpoint commented:
You're all acting like children.

RemieRichards commented:
@Steelpoint No, goof is being a moron as per usual threatening his memes, I'm explaining his memes will get him nowhere, and you're adding nothing to the discussion.

dannno commented:
Goofball, everything you do is so petty and transparent
everyone sees it you know

JJRcop commented on code/modules/clothing/under/miscellaneous.dm:
```diff
@@ -158,13 +165,15 @@
 	icon_state = "owl"
 	item_color = "owl"
 	can_adjust = 0
+	identity_name = "The Owl"
```
Yes, THE Owl, superhero

JJRcop commented on code/controllers/subsystem/jobs.dm:
```diff
@@ -392,7 +392,32 @@ var/datum/subsystem/job/SSjob
+		var/datum/mind/H_mind = H.mind
+		H_mind.assigned_role = rank
+
+		if(H.voiceprint && (!joined_late || world.time - round_start_time <= config.latejoin_identity_retcon))
```
Where, the `world.time - round_start_time <= config.latejoin_identity_retcon`?
ATM it's H.voiceprint is true AND (they aren't a latejoin OR the latejoin backfill time hasn't expired)

Iamgoofball commented:
I'm aware

JJRcop commented on code/datums/mind.dm:
```diff
@@ -42,14 +42,18 @@
 	var/special_role
 	var/list/restricted_roles = list()
 
-	var/datum/job/assigned_job
```
It isn't used, anywhere... It's not even set. It's an orphaned var

JJRcop commented on code/game/machinery/computer/crew.dm:
```diff
@@ -194,21 +198,25 @@ var/global/datum/crewmonitor/crewmonitor = new
+							tracking_pos = null
 
-						results[++results.len] = list(name, assignment, ijob, life_status, dam1, dam2, dam3, dam4, area, pos_x, pos_y, H.can_track(null))
+						results[++results.len] = list(name, assignment, ijob, life_status, dam1, dam2, dam3, dam4, area, pos_x, pos_y, H.can_track(null), tracking_pos)
```
This is strictly to allow the AI to click a name if they have tracking beacon enabled and track them

JJRcop commented on code/modules/clothing/suits/miscellaneous.dm:
```diff
@@ -117,6 +121,7 @@
 	item_state = "chickensuit"
 	body_parts_covered = CHEST|ARMS|GROIN|LEGS|FEET
 	flags_inv = HIDESHOES|HIDEJUMPSUIT
+	identity_name = "chicken?"
```
Not strictly this but I have put question marks in names using the identity manager and it worked then, i can always test this

ChemicalRascal commented:
He does have a point, though, just because this is substantially different doesn't make it _bad_, and players do not expert game designers make purely by being players. Sure he's a knob, but he is right in noting that the cries of "this sucks plz kill" are premature.

Razharas reviewed (commented):
Im reviewing this

JJRcop commented on code/modules/events/wizard/shuffle.dm:
```diff
@@ -61,7 +61,9 @@
-		H.real_name = mobnames[mobnames.len]
+		var/mobname = mobnames[mobnames.len]
+		H.real_name = mobname[1]
+		H.voiceprint = mobname[2]
```
Line 52 deals with getting all the mob's name and voiceprint, and indexing them in a list, this stuff is actually giving out the shuffled names and voiceprints.

Delimusca commented:
you could fix the problem of radio unknowns by tying broadcasts to area and/or department radio.
I.E _Engineer in Atmospherics says "who took all the pipes?"_
or _Unknown in Maintenance says "o-oOPS!"_

factoryman942 commented:
What if you made a new flag-thing for certain masks or something which conceals/muffles radio voice, meaning that people have a % chance to recognise the voice if they've heard stuff said by the person before just over local talk or voice recorder?

_it'd make a use for voice recorders too apart from reading erotica_

PKPenguin321 commented:
Despite my hatred for this PR I would still like my question answered:
>How the fuck will the radio work with this? I feel like it will become unusable

JJRcop commented:
@PKPenguin321 Radio works fine, but if you don't know the person they show up as Unknown, if you do know the person they will show up as what you remember them by. This also works if you rename them, and I'm going to make renaming very simple in my next commit where you just click the name and get a prompt to rename them to whatever.

PKPenguin321 commented:
You should make it so the radio always shows names if this does go through, at least that way the radio won't be ruined

Grazyn commented:
@JJRcop I have a couple questions, let's say you know the name and face of a person, but you've never heard them speak, do they still show as Unknown on the radio? And similarly, you hear someone speak on the radio but you can't see them, then you find them and examine them in 50 seconds, does the voiceprint get linked with the face even though you have no way to know they were the ones speaking earlier?,

optimumtact commented:
![part](https://cloud.githubusercontent.com/assets/2544323/22819336/846d7c56-efd6-11e6-9c3f-07ca8611bdc8.jpg)

JJRcop commented:
@Razharas Any status on the review?

JJRcop commented on code/__HELPERS/time.dm:
```diff
@@ -1,12 +1,12 @@
-/proc/gameTimestamp(format = "hh:mm:ss") // Get the game time in text
-	return time2text(world.time - timezoneOffset + 432000 - round_start_time, format)
+/proc/gameTimestamp(timestamp = world.time, format = "hh:mm:ss") // Get the game time in text
+	return time2text(timestamp - timezoneOffset + 432000 - round_start_time, format)
```
proc is only used in worldtime2text atm and I already adjusted that

SouDescolado commented:
To be honest, I always think SS13 is a glorified "Secret Hitler" or "Detective". Hell, its even like "Town of Salem". Also not knowing the names of everyone takes away the disguise part of the game (Where you have to steal someone else ID, for example). It removes the trickery aspect.

I'm all in for this if it's a roundtype. Like "ID system was offline, everyone is unknown on this round unless you talk to them". Mix it with syndie rounds and voila. But not a mandatory thing where EVERY round you would need to know everyone, get everyone to know you, and THEN pretend to be someone else and fuck everyone over.

Haevacht commented:
The game you are thinking of is called Mafia. The original is fuckign Mafia.

SouDescolado commented:
Thanks @Haevacht . Pardon my lack of knowledge. But the point stands.

Shadowlight213 commented:
How can it be secret hitler when everyone knows hg has the distinguished title of "Literally hitler"?
Not much of a secret.

ChemicalRascal commented:
That's not a bad idea, making it a round type (or should I suggest a round type modifier). Certainly, that'd allow for it to be tested in a non-meme way, and we could see how people treat the system when they're actually familiar with it.

RemieRichards commented:
I'm genuinely sorry it took me so long to link this
https://www.youtube.com/watch?v=Z4ELcAmdqoY

JJRcop commented:
@SouDescolado I'm confused as to how not knowing everyone takes away from disguising, I thought it would make disguising easier.

Ninja137 commented:
It makes sense, the code is presumably not dogshit, and it shits on the metafag bullshit that's been plagueing the station for literal years, even if by only a little. Merge it.

RemieRichards commented:
@JJRcop Your extra week runs out today, but I'll let you have the friday if you need it.

JJRcop commented:
@RemieRichards I have some changes I am making due to feedback, it was reviewed by @Razharas but he didn't finish his GitHub review and just told me what I should change over IRC. He said he doesn't see any major problems with the code in general, so this PR is not dead yet.

RemieRichards commented:
The code my not be, but as I said a week ago, your supporters are in the minority.

JJRcop commented:
Your original comment for closing this PR was
>if it's not reviewed, nobody cares enough and I'll close it.
It's been reviewed, and I'm making changes. People do care about this, and we're moving forward from this week of nothing, so could we forget about the time limit?

There was a lack of communication between me and Raz so I didn't start acting on the feedback until now.

RemieRichards commented:
From the comments here and the forums:
8 Are in favour
14 Are Against
(Everyone is counted once, people who changed their mind (nobody?) aren't counted twice)

I'm tired of this, @Cheridan / @KorPhaeron (you still the design lead?), you decide.

Doppeltongue commented:
This seems like a massive buff to all antags. It makes cross-department co-operation more difficult and tedious while leaving team antags just fine and makes it easier for solo antags to murder people without sec being able to help. Also, I don't envy sec who now has to deal with an entire station full of Unknowns.

kevinz000 commented:
this doesn't belong on a stun combat orientated game. would be nice if we somehow moved away from it and this was tried again.

optimumtact commented:
@RemieRichards we're running a in game poll right now, should take about a week.

Then based on that we will either extended test merge this, or stop proceeding with it.

Then after about a month we would run another poll

RemieRichards commented:
Whatever, as far as I'm concerned it's had it's chances.

SouDescolado commented:
I just think it shouldn't be mandatory. But as a round type modifier I think it would be fine.

Razharas commented:
I think its idea that can work but currently made poorly(menu fiddling for renmaing and horrid default names for unknown people are just bad, also maybe we can add some info about unknowns in common channel to avoid uniformity spam)
And then we can finally proceed to nerfing antag deadliness and maybe whole game deadliness in general

RemieRichards commented:
>nerfing the deadliness

That seems contrary to what's wanted, People want deadlier combat with less/no stuns, we should be heading towards that, not away.

Steelpoint commented:
How this system works need to be very clearly, and easily explained to everyone who joins a game.

This is a system that will push new players away if they don't understand it.

Even if you have to spawn a paper in their hand that explains how the system works.

JJRcop commented:
Thanks for that @Steelpoint, definitely good to have.

Razharas commented:
RR stuns are the most deadly of them all coz you can kill stunned person with a spoon, which bascially makes any stun a one hit kill weapon
Nerfing stuns is nerfing deadliness in itself

peoplearestrange commented:
Like this could be interesting.
 I think security should probably know the names of all the heads or command staff. Also if its not already the case, all command staff should know each other. That would make sense.

Steelpoint commented:
I think this system already has the heads of staff know one another.

Good point on Security knowing the names of all the heads of staff, one less headache to worry about.

JJRcop commented:
Mmm with the way I have it set up right now, I don't support specific roles only knowing other roles one-way, and I think that would make things too complicated. So if I wanted sec to know all heads of staff, all heads of staff would know sec too.

I guess it isn't that big of a deal... So just officers, or all of sec? Maybe just detectives?

Robustin commented:
A curious system but its ultimately not going to work.

JJRcop commented:
I'm going to leave security out of knowing the heads for now, it's gonna be a config file.

Spacedong commented:
>he still thinks this can get merged
![good_dhit](https://cloud.githubusercontent.com/assets/12958054/23145949/7376121a-f7a0-11e6-9a6d-3b27f41d3251.png)

optimumtact commented:
Would you be interested in further testing, and possibly permanent merging of https://github.com/tgstation/tgstation/pull/23693 ? (link)

Yes	167	56%	
No	133	44%

This needs to be brought to a ready state so we can have it merged for a longer period. I'm thinking two weeks at least.

JJRcop commented:
Current in-progress changes:

>examine needs to be better
click on names in chat to rename them, don't use the identity manager for this, but keep it around still
list lengths should be defines
move global regex vars to be static vars in the procs (and probably remove /proc/consonants)
raz doesn't like var/_these_vars
add friends??
~~*IDs have the picture on them...??? (i don't like this idea, not going to do it for right now but will reconsider in the future)*~~
limited range for faceprint gathering?
config file for pre-knowns
add accents
some form of help
potential garbage issues for identity manager

SouDescolado commented:
Add something like #24323 so you can recognize people by smell 👃

JJRcop commented:
@SouDescolado out of scope

RemieRichards commented:
I will be incredibly disappointed if this succeeds.

PKPenguin321 commented:
^

Just because the code is good doesn't mean the feature is good, I really don't like this in concept or execution

SouDescolado commented:
I'll say it once more before the code is scrapped: Make it a round type.

Dorsisdwarf commented:
Its interesting how the general server's opinion of this seems to slowly be shifting towards positive

Steelpoint commented:
It usually takes weeks for a new system or feature to be fully evaluated by the community in game.

RemieRichards commented:
@Dorsisdwarf they could only be saying they wish to test it more, I don;t like how the poll is ambiguous:
>Would you be interested in further testing, and possibly permanent merging of #23693 ? (link)

You can only say "yes" or "no", you can't say "yes to more tests", or "yes to merge"

Dorsisdwarf commented:
I said it was "interesting", not "a ringing endorsement"

RemieRichards commented:
I know? I never challenged that, I challenged what you FOUND interesting, this "trend towards positivity", which I've just explained is ambiguous and may actually be "trend towards more tests"

ChemicalRascal commented:
Well that clearly indicates there's interests in, at a bare minimum, more
testing, then.

Sou is by far and away the unheard genius here. This would be excellent as
a round type (I personally think it'd be better as a round modifier than a
round type but tomato/tomato).

Remie, you're not really in a position to challenge what someone finds
interesting. You're not inside their heads. Sometimes I find wallpaper
patterns interesting.

On Thu., 23 Feb. 2017, 23:41 Remie Richards, <notifications@github.com>
wrote:

> @Dorsisdwarf <https://github.com/Dorsisdwarf> they could only be saying
> they wish to test it more, I don;t like how the poll is ambiguous:
>
> Would you be interested in further testing, and possibly permanent merging
> of #23693 <https://github.com/tgstation/tgstation/pull/23693> ? (link)
>
> You can only say "yes" or "no", you can't say "yes to more tests", or "yes
> to merge"
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/tgstation/tgstation/pull/23693#issuecomment-281981831>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABpDZfL3BaA0sfdsQzQd_sXjvleF26Suks5rfX6NgaJpZM4L3KrT>
> .
>

RemieRichards commented:
Why are you all so dense.
I am not challenging that he finds it interesting, I'm saying he's interested in something that <i>may not even be true</i>, it's fine for him to be interested in wrongness, but that doesn't make it not wrong.

Dorsisdwarf commented:
What I mean is that when it was posted, everyone was shitting all over it.

Now people seem more interested in it, despite it not seeming to have changed much, and I was interested about why that could be.


No idea why people are getting in fights about this it was an idle observation while I wait for KSP to download

RemieRichards commented:
>Now people seem more interested in it

THIS is my point, <i>this is not fact</i> it's ambiguous because of the question the poll posed, "Test More" and "I Like this" are TIED TOGETHER, thus you cannot say "Oh wow! look at all the people who like this!

Test More + I Like = Yes Result
I Like = Yes Result - Test More
I Like = 56% - ??%

?? Could be 2%, it could be 30% it could even be the entire 56% (unlikely however as some people do genuinely want this, such as JJ himself), so the break down of "56% like this" is wrong, you can only say "44% of people are dead against this" (although technically the no result has the same problem just inverted, however if people like something the odds of them saying no to seeing more of it is almost 0)

ChemicalRascal commented:
There aren't really "people" getting into fights -- Remie is the only person who's upset here, and apparently I misread something but *hands up don't shoot*

In all honesty I only commented because Remie is the most vocally negative person in the thread, constantly reminding the thread of their pending disappointment -- which is fine, but they're also a maintainer, and I'm worried they'll close the PR prematurely because they overestimate general disapproval due to their own intense disapproval. I think this PR would be a lot less... emotional if a more dispassionate maintainer was involved.

Idk, enjoy green mini spessmen

Dorsisdwarf commented:
I said *interested* in it not pro-identity, you dont need to bring out statistics.

![this is the war room](https://68.media.tumblr.com/b290c5ceebd8fbab1a426662d929d53f/tumblr_mi6e78uto61qepf8yo1_500.gif)

RemieRichards commented:
@ChemicalRascal Your fears are unfounded, I'm not going to go against what we're doing (which, from the results of the poll, is more tests), if you'll look up I declared this PR dead twice only to extend it once for JJ, and to concede to some polls and tests the second time.

If a final verdict is given, I will uphold it, regardless of what I feel.
We have an unfortunate attitude that "if the code's good, it gets merged" regardless if the concept is rotten to the core, and I'm just stating my opinion on that (as have others)

Dorsisdwarf commented:
1) Remie is a good maintainer, and I doubt they'd ever shut down a project just because they're opposed

2) They're right about us having a tendency to merge almost anything "just because it's finished"

ChemicalRascal commented:
But it remains that you so quickly called the PR dead, twice, in the first place, and you were the only voice to ever do so. At a bare minimum, that looks like a certain eagerness to shut it down.

ChemicalRascal commented:
I mean, I'm happy to be wrong, and I'm not trying to throw up my hands and say "Remie is Hitler!", I just have concerns.

RemieRichards commented:
I will however take this opportunity to apologise for being a bit too zealous.

@ChemicalRascal the PR hadn't moved for a while each time I did so, PRs like these sit for literal months with no action and are finally killed off for being "stale", ironically because of this I may have CAUSED it's eventual merger simply by kick starting the process again by claiming it dead.

ChemicalRascal commented:
And it's important to note that, while I agree we do happen to merge lot (I'd argue on small features rather than significant stuff like this), from what I've seen bad stuff gets reverted just as quickly.

PKPenguin321 commented:
>bad stuff gets reverted just as quickly.

Hahaha

JJRcop commented:
I see the poll as people who are interested in seeing how further testing goes vs people who are not interested in testing and want this PR closed immediately, and I believe that was the intent of the poll, but I am not certain as I didn't make the poll.

I will say that, given a scenario where the majority agrees with the results of further testing and says "yeah merge this", it gets merged, then a month down the line those people get super sick of it, I will make my effort to remove it, since I made it.

I've been around this codebase on and off for 3 years, I am not just going to leave forever like others have done, and with this system, merged or not, I will be around here more often anyway.

Also, as I have said before, I am super open to feedback for this system, and I want people to say how it could be better. I do not agree with the round type that @SouDescolado suggested, and will not be making changes in that direction at this point in time.

Commit:
Puts roundstart departments in a config file
this can be found in config/preknown_jobs.txt, where you define "groups", and put job titles in those groups. At roundstart, the groups get parsed, and for every group your job title is in, you will be preset with all jobs in that group.
There are two special groups, Famous and Friendly. Roles in Famous will automatically be preset into everyone, while roles in Friendly will automatically have roles preset into them.
New mind/preknown_identity proc for this purpose which presets both voiceprint and faceprint of target mob, and links them together if applicable
Moved link and unlink procs out of the identity_manager into mind

PKPenguin321 commented:
>I am super open to feedback for this system

Can you make it a config if it isn't already?

Iamgoofball commented:
@PKPenguin321 
no

JJRcop commented:
@PKPenguin321 You can put all jobs into one group in preknown_jobs.txt now, effectively disabling the system in most cases. In a future commit I will also allow you to provide a negative time to the LATEJOIN_IDENTITY_RETCON config option, to represent true infinity. (as opposed to 1e31)

As some examples for the times the identity system would come into effect with those settings, is non-crewmembers, as in nuke ops, wouldn't know the crew's names, and putting an alien brain inside a human body, then cloning that body, would cause them to not know anybody's name, this actually applies for all minds that are not assigned a job when they join, (or are not equipped with their gear)

peoplearestrange commented:
@RemieRichards Im curious to what you dislike about the PR? Can you explain it simply to me from a feature POV for my simple bluefox mind.

To me it seems like a feature that could add paranoia and confusion, but the reality of that might be its more of an annoying inconvenience rather than a good game mechanic.

RemieRichards commented:
@peoplearestrange you're right, those are the promises it makes, but it won't fulfill them, all this would lead to is a tedious 10 minutes at the start of the round running around clicking people to get their info, or worse, just completely ignoring people altogether (which is I assume counter to this, since people say that you'll "have to talk to people!")

cacogen commented:
>commented 4 days ago

seems dead 

but speculating about what effect this will have on the round is pretty dumb when you could just testmerge it

RemieRichards commented:
It HAS been testmerged.

cacogen commented:
haha my guy

JJRcop commented:
I am working on it

francinum commented:
It WAS testmerged, and that ended horribly.

On 2/28/17, JJRcop <notifications@github.com> wrote:
> I am working on it
>
> --
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/tgstation/tgstation/pull/23693#issuecomment-283113187

optimumtact commented:
You need to get this to a mergeable state so we can actually proceed with the test merges.

Commit:
You can click on names in chat to rename their attached voiceprint
Tried to make the links not blue but failed, will try again later
Changed some variables from _variable to new_variable
Made some regexes static vars inside their procs
Changed some list new() lengths to be defines

Iamgoofball commented:
Wrong button?

JJRcop commented:
Gonna reopen when I finish complying with oranges' request.

Iamgoofball commented:
Don't

Reopen it now, PRs don't update when closed

JJRcop commented:
Oranges actually wanted me to make a new PR instead because of the fact that I'm not getting any more useful feedback in this PR and it's got so many unhelpful comments.

Iamgoofball commented:
Ok
